### PR TITLE
Add CI files

### PR DESCRIPTION
**Pull request by NamsooCho**

### CODE REVIEW

File: scripts/bootstrap-osx.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env bash
+
+# used in travis to download & install libsodium
+
+brew update
+brew install libsodium
```
Comment (kpp) [inline scripts/bootstrap-osx.sh]: We don't need it

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  - cargo doc
+
+after_success:
+  # measure code coverage and upload to coveralls.io
+  - |
+    if [[ "$TRAVIS_OS_NAME" == "linux" ]] && [[ "$TRAVIS_RUST_VERSION" == stable ]]
+    then
+      cargo install cargo-travis || echo "cargo-travis has been already installed"
+      export PATH=$HOME/.cargo/bin:$PATH
+      cargo coveralls
```
Comment (kpp) [inline .travis.yml]: we don't need it

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        ./scripts/bootstrap-osx.sh
+    fi
+
+script:
+  - cargo build --verbose
+  - cargo test --verbose
+  - |
+    if [[ "$TRAVIS_RUST_VERSION" == nightly ]]
+    then
+        cargo build --features "clippy" \
```
Comment (kpp) [inline .travis.yml]: we don't need it

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+addons:
+  apt:
+    packages:
+      - libcurl4-openssl-dev
+      - libelf-dev
+      - libdw-dev
+      - binutils-dev
+      - cmake
+    sources:
+      - kalakris-cmake
```
Comment (kpp) [inline .travis.yml]: we don't need it

File: .gitlab-ci.yml [hunk 5]
```diff
@@ -0,0 +1,175 @@
+variables:
```
Comment (kpp) [inline .gitlab-ci.yml]: we don't need gitlab ci

File: .travis.yml [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+  - |
+    if [[ "$TRAVIS_OS_NAME" == "linux" ]]
+    then
+        ./scripts/verify-commit-messages.sh "$TRAVIS_COMMIT_RANGE" \
+        && ./scripts/bootstrap-ubuntu-14-04.sh
+    fi
+  - |
+    if [[ "$TRAVIS_OS_NAME" == "osx" ]]
+    then
+        ./scripts/bootstrap-osx.sh
```
Comment (kpp) [inline .travis.yml]: we don't need it

File: scripts/bootstrap-osx.sh (hunk 1 above)
Comment (NamsooCho) [reply scripts/bootstrap-osx.sh]: = note: /usr/bin/ld: cannot find -lsodium
          collect2: error: ld returned 1 exit status

Comment (kpp) [reply scripts/bootstrap-osx.sh]: Ah sorry. I did not see you enabled osx builds

File: .travis.yml (hunk 6 above)
Comment (NamsooCho) [reply .travis.yml]: done

File: .travis.yml [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    fi
+
+script:
+  - cargo build --verbose
+  - cargo test --verbose
+  - cargo doc
```
Comment (kpp) [inline .travis.yml]: We don't need to document it right now.

Comment (NamsooCho) [reply .travis.yml]: done

**Pull request merged**